lib/src/trace.dart

```diff
@@ -285,7 +285,10 @@ class Trace implements StackTrace {
         var library = frame.library.replaceAll(_terseRegExp, '');
         return new Frame(Uri.parse(library), null, null, frame.member);
       }).toList();
-      if (newFrames.length > 1 && newFrames.first.isCore) newFrames.removeAt(0);
+
+      if (newFrames.length > 1 && predicate(newFrames.first)) {
```

Comment:
Not really related to this CL but I don't quite follow why we only fold the first frame.
Perhaps we should fold all but the last frame?
That way, we would be stable if 
Trace.parse() and Trace.toString() are called repeatedly.
Otherwise I worry we will strip off one frame each time which is a bit surprising.